### PR TITLE
feat(cli): implement HTML formatter

### LINKED ISSUE #389: HTML Format
**User story.**
As a CLI user, I want to run the CLI, output information about failures, and put that HTML on S3 so folks can see a report of their errors and warnings. Maybe this will help them fix the various issues.

**Describe the solution you'd like**

Rubocop have a good HTML format: https://www.rubocop.org/en/stable/formatters/

Check out ESLint too.

### PR DESCRIPTION
**Pull request by P0lip**
Closes #389 

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Screenshots**

localhost_63342_spectral_test html__ijt=d1nrv39kkdr5jm5tebd0jvn5s5

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,9 +43,10 @@
     "inline-version": "./scripts/inline-version.js",
     "lint.fix": "yarn lint --fix",
     "lint": "tsc --noEmit && tslint 'src/**/*.ts'",
+    "copy.html-templates": "copyfiles -u 1 \"./src/formatters/html/*.html\" \"./dist/\"",
```
Comment (philsturgeon) [inline package.json]: Is there any other way to go about this? Is this standard? It feels a little clunky to me, but maybe its fine?

Comment (P0lip) [reply package.json]: > Is there any other way to go about this? Is this standard?

What do you mean? We can place it directly in `prebuild`, no need to have this extra script. I just wanted to make is slightly more readable.

Comment (philsturgeon) [reply package.json]: I just meant... this feels like an "asset pipeline" sort of thing, I wonder if there are more common ways people handle "not typescript" in typescript libraries than copying bits over. Nevermind :D

**Comment by P0lip** [comment]
@philsturgeon could you re-approve? Had to normalize line endings for Windows (it used CRLF instead of LF).

**Pull request merged**